docs/source/en/package_reference/inference_endpoints.md

```diff
@@ -0,0 +1,48 @@
+# Inference Endpoints
+
+Inference Endpoints offers a secure production solution to easily deploy any `transformers`, `sentence-transformers`, and `diffusers` models from the Hub on a dedicated and autoscaling infrastructure managed by Hugging Face.
+
+An Inference Endpoint is built from a model from the [Hub](https://huggingface.co/models). When an Endpoint is created, the service creates image artifacts that are either built from the model you selected or a custom-provided container image. The image artifacts are completely decoupled from the Hub source repositories to ensure the highest security and reliability levels.
+
+Inference Endpoints support all of the `transformers`, `sentence-transformers` and `diffusers` tasks as well as custom tasks not supported by `transformers` yet like speaker diarization.
+
+In addition, Inference Endpoints gives you the option to use a custom container image managed on an external service, for instance, [Docker Hub](https://hub.docker.com/), [AWS ECR](https://aws.amazon.com/fr/ecr/?nc1=h_ls), [Azure ACR](https://azure.microsoft.com/de-de/products/container-registry/), or [Google GCR](https://cloud.google.com/artifact-registry?hl=de).
+
+<Tip>
+
+This page is only a reference for `huggingface_hub`'s integration with Inference Endpoints. For more information about the Inference Endpoints product, check out its [official documentation](https://huggingface.co/docs/inference-endpoints/index).
+
+</Tip>
+
+Inference Endpoints can be fully managed via API. The endpoints are documented with [Swagger](https://api.endpoints.huggingface.cloud/).
```

Comment:
Could be interesting to add here that the `InferenceClient` is a python implementation of that swagger